Java/PenPaper/Paper.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Created by andrew on 03.12.16.
+ */
+public class Paper {
+    private int maxSymbols;
+    private int symbols;
+    private StringBuffer content;
+
+    public Paper(int maxSymbols) {
+        content = new StringBuffer();
+        this.maxSymbols = maxSymbols;
+        this.symbols = 0;
+    }
+
+    public Paper() {
+        this(4096);
+    }
+
+    public int getMaxSymbols() {
+        return maxSymbols;
+    }
+
+    public int getSymbols() {
+        return symbols;
+    }
+
+    public void addContent(String message) throws OutOfSpaceException {
+        int total = content.length() + message.length();
+
+        if ( total > maxSymbols ) {
+            content.append(message.substring(0, maxSymbols));
```

Comment:
I guess that you don't have to append substring from 0 to `maxSymbols`. In such case you can get string with` length == maxSymbols` and that will cause an overhead